src/client/Microsoft.Identity.Client/Platforms/Android/AndroidBrokerHelper.cs

```diff
@@ -0,0 +1,488 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Security.Cryptography.X509Certificates;
+using AndroidNative = Android;
+using Android.Accounts;
+using Android.App;
+using Android.Content;
+using Android.OS;
+using Android.Content.PM;
+using Android.Util;
+using Java.Security;
+using Java.Util.Concurrent;
+using OperationCanceledException = Android.Accounts.OperationCanceledException;
+using Permission = Android.Content.PM.Permission;
+using Signature = Android.Content.PM.Signature;
+using Microsoft.Identity.Client.Core;
+using Microsoft.Identity.Client.OAuth2;
+using Microsoft.Identity.Client.Internal.Broker;
+using Microsoft.Identity.Client.Utils;
+using Microsoft.Identity.Json.Linq;
+
+namespace Microsoft.Identity.Client.Platforms.Android
+{
+    internal class AndroidBrokerHelper
+    {
+        private const string RedirectUriScheme = "msauth";
+        private const string BrokerTag = BrokerConstants.Signature;
+        public const string WorkAccount = "com.microsoft.workaccount.user.info";
+
+        private readonly Context _androidContext;
+        private readonly AccountManager _androidAccountManager;
+        private readonly ICoreLogger _logger;
+
+        public AndroidBrokerHelper(Context androidContext, ICoreLogger logger)
+        {
+            _androidContext = androidContext ?? throw new ArgumentNullException(nameof(androidContext));
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+
+            _logger.Verbose("Getting the Android context");
+            _androidAccountManager = AccountManager.Get(_androidContext);
+        }
+
+        public bool CanSwitchToBroker()
+        {
+            string packageName = _androidContext.PackageName;
+
+            // ADAL switches broker for following conditions:
+            // 1- app is not skipping the broker
```

Comment:
>skipping the broker [](start = 29, length = 19)

what does #1 mean?

Reply:
it is checking if the broker app in installed. Updated